hack/tools/extension-generator/options.go

```diff
@@ -36,6 +38,7 @@ func (o *Options) AddFlags(flags *flag.FlagSet) {
 	flags.StringVar(&o.ExtensionOCIRepository, "extension-oci-repository", "", "URL to OCI image containing the extension chart")
 	flags.StringVar(&o.AdmissionRuntimeOCIRepository, "admission-runtime-oci-repository", "", "OPTIONAL: URL to OCI image containing the admission runtime chart")
 	flags.StringVar(&o.AdmissionApplicationOCIRepository, "admission-application-oci-repository", "", "OPTIONAL: URL to OCI image containing the admission application chart")
+	flags.BoolVar(&o.InjectGardenKubeconfig, "inject-garden-kubeconfig", false, "OPTIONAL: When set, the `.spec.deployment.extension.injectGardenKubeconfig: true` field is added to the generated extension")
```

Comment:
This tool tries to refrain from binding API fields to command line flags, see https://github.com/gardener/gardener/pull/11593#discussion_r1982240280. Instead, please add this field to the opinionated provider-extension resource calculation [here](https://github.com/gardener/gardener/blob/836fd717fce422362b04b3b582e9b22f9f40761c/hack/tools/extension-generator/extension.go#L132).

Reply:
Sorry, from our offline discussion I was left with the impression that we could expose this option in case other extensions, apart from providers (responsible for worker resource), want to access the garden cluster. For provider extensions I think it's better to still rely on the defaulting done in [here](https://github.com/gardener/gardener/blob/ef6b8e77e351199aaa6a14ac332750f6ec044fe2/pkg/operator/webhook/defaulting/extension/handler.go#L24-L37)

Reply:
Would you remove it again @plkokanov?

Reply:
Yes, sorry for the delay. Got side tracked with some other tasks. Will try to update the PR a bit later today.